internal/controller/managedcluster_controller.go

```diff
@@ -269,9 +272,47 @@ func (r *ManagedClusterReconciler) Update(ctx context.Context, l logr.Logger, ma
 		Message: "Helm chart is valid",
 	})
 
+	cred := &hmc.Credential{}
+	err = r.Client.Get(ctx, client.ObjectKey{
+		Name:      managedCluster.Spec.Credential,
+		Namespace: managedCluster.Namespace,
+	}, cred)
+	if err != nil {
+		apimeta.SetStatusCondition(managedCluster.GetConditions(), metav1.Condition{
+			Type:    hmc.CredentialReadyCondition,
+			Status:  metav1.ConditionFalse,
+			Reason:  hmc.FailedReason,
+			Message: fmt.Sprintf("Failed to get Credential: %s", err),
+		})
+		return ctrl.Result{}, err
+	}
+
+	if cred.Status.State != hmc.CredentialReady {
```

Comment:
we will have this check in admission for the initial managedcluster creation. but after it passed validation initially (e.g. the credential was ready when managedcluster was created, but got invalid some time later), I don't think it makes sense to fail the reconcile anymore